test/unit/assorted/server_selection_spec_helper.js

```diff
@@ -102,104 +102,84 @@ function executeServerSelectionTest(testDefinition, testDone) {
-    topologyDescription.servers.forEach(server => {
-      const serverDescription = serverDescriptionFromDefinition(server, seedData.hosts);
-      topology.serverUpdateHandler(serverDescription);
-    });
+  // default to serverSelectionTimeoutMS of `100` for unit tests
+  try {
+    const server = await topology.selectServer(selector, { serverSelectionTimeoutMS: 50 });
 
-    let selector;
-    if (testDefinition.operation === 'write') {
-      selector = ServerSelectors.writableServerSelector();
-    } else if (testDefinition.operation === 'read' || testDefinition.read_preference) {
-      try {
-        const readPreference = readPreferenceFromDefinition(testDefinition.read_preference);
-        selector = ServerSelectors.readPreferenceServerSelector(readPreference);
-      } catch (e) {
-        if (testDefinition.error) return done();
-        return done(e);
-      }
-    } else {
-      return done(
-        new Error('received neither read nor write, and did not receive a read preference')
-      );
+    if (testDefinition.error) throw new Error('Expected an error, but found none!');
+    if (expectedServers.length === 0 && server !== null) {
+      throw new Error('Found server, but expected none!');
     }
 
-    // expectations
-    let expectedServers;
-    if (!testDefinition.error) {
-      expectedServers = testDefinition.in_latency_window.map(s =>
-        serverDescriptionFromDefinition(s)
-      );
-    }
+    const selectedServerDescription = server.description;
 
-    // default to serverSelectionTimeoutMS of `100` for unit tests
-    topology.selectServer(selector, { serverSelectionTimeoutMS: 50 }, (err, server) => {
-      // are we expecting an error?
-      if (testDefinition.error) {
-        if (!err) {
-          return done(new Error('Expected an error, but found none!'));
-        }
+    const expectedServerArray = expectedServers.filter(
+      s => s.address === selectedServerDescription.address
+    );
 
-        return done();
-      }
+    if (!expectedServerArray.length) {
+      throw new Error('No suitable servers found!');
+    }
 
-      if (err) {
-        // this is another expected error case
-        if (expectedServers.length === 0 && err instanceof MongoServerSelectionError) return done();
-        return done(err);
-      }
+    if (expectedServerArray.length > 1) {
+      throw new Error('This test does not support multiple expected servers');
+    }
 
-      if (expectedServers.length === 0 && server !== null) {
-        return done(new Error('Found server, but expected none!'));
+    for (const [prop, value] of Object.entries(expectedServerArray[0])) {
+      if (prop === 'hosts') {
+        // we dynamically modify this prop during sever selection
+        continue;
       }
+      expect(selectedServerDescription[prop]).to.deep.equal(
+        value,
+        `Mismatched selected server "${prop}"`
+      );
+    }
+    return;
+  } catch (err) {
+    // are we expecting an error?
```

Comment:
tbh I thought this was intended to be a question for the reviewers 😅 

I don't feel strongly but maybe making it a statement instead of a question would be a little more clear